docs/extend/plugins.md

```diff
@@ -1,87 +1,272 @@
 <!--[metadata]>
 +++
-title = "Extending Engine with plugins"
-description = "How to add additional functionality to Docker with plugins extensions"
-keywords = ["Examples, Usage, plugins, docker, documentation, user guide"]
+aliases = [
+"/engine/extend/"
+]
+title = "New Plugin System"
```

Comment:
it's too bad git doesn't handle this well. the old file was actually `git mv` to `legacy_plugins.md` .  see note on that file.

I'm not against creating a new file here, but if we do that, we should move the old file back to `plugins.md`. Either way we have a link maintenance task. We need to go through and update all the links to `plugins.md` to `plugins-overview.md` or fix broken links.

@thaJeztah perhaps you see something that I don't ?